src/App.jsx

```diff
@@ -14,45 +15,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABETICALLY = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+
+function getPreparedGoods(goods, { typeOfSort, isReversed }) {
+  let preparedGoods = [...goodsFromServer];
+
+  if (typeOfSort) {
+    preparedGoods = preparedGoods.sort((good1, good2) => {
+      switch (typeOfSort) {
+        case SORT_BY_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [typeOfSort, setTypeOfSort] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleEffects = getPreparedGoods(
+    goodsFromServer, { typeOfSort, isReversed },
+  );
+
+  const handleSort = (sortType) => {
+    setTypeOfSort(sortType);
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${SORT_BY_ALPHABETICALLY === typeOfSort ? '' : 'is-light'}`}
```

Comment:
`className={`button is-info ${SORT_BY_ALPHABETICALLY !== typeOfSort && 'is-light'}`}`. Fix everywhere